src/basic/namespace-util.c

```diff
@@ -231,3 +231,25 @@ int userns_acquire(const char *uid_map, const char *gid_map) {
         return TAKE_FD(userns_fd);
 
 }
+
+bool in_same_namespace(pid_t pid1, pid_t pid2, namespace_type ns)
+{
+        const struct namespace_info *nsinfo = &namespace_info[ns];
+        char ns_path[STRLEN("proc//") + DECIMAL_STR_MAX(pid_t) + strlen(nsinfo->proc_path)];
+        struct stat ns_st1, ns_st2;
+
+        xsprintf(ns_path, "/proc/" PID_FMT "/%s", pid1, nsinfo->proc_path);
+        if (stat(ns_path, &ns_st1))
+                return -errno;
+
+        xsprintf(ns_path, "/proc/" PID_FMT "/%s", pid2, nsinfo->proc_path);
+        if (stat(ns_path, &ns_st2))
+                return -errno;
+
+        /* processes are in the same namespace */
+        if ((ns_st1.st_dev == ns_st2.st_dev) &&
+            (ns_st1.st_ino == ns_st2.st_ino))
+                return true;
+
+        return false;
```

Comment:
we have a helper for that, since we have the same expression quite often. i.e. just do:

```c
return stat_inode_same(&ns_st1, &ns_st2);
```